markdown/ch1.markdown

```diff
@@ -17,28 +17,35 @@ In this case, we want to highlight NumPy's vectorization and broadcasting rules,
 ```python
 def rpkm(counts, lengths):
     """Calculate reads per kilobase transcript per million reads.
+
     RPKM = (10^9 * C) / (N * L)
 
-    Where:  
-    C = Number of reads mapped to a gene  
-    N = Total mapped reads in the experiment  
+    Where:
+    C = Number of reads mapped to a gene
+    N = Total mapped reads in the experiment
     L = Exon length in base pairs for a gene
 
-    counts: 2D numpy ndarray (numerical)
+    Parameters
+    ----------
+    counts: array, shape (N_genes, N_samples)
         RNAseq (or similar) count data where columns are individual samples
         and rows are genes.
-    lengths: list or 1D numpy ndarray (numerical)
+    lengths: array, shape (N_genes,)
         Gene lengths in base pairs in the same order
         as the rows in counts.
-    """
 
+    Returns
+    -------
+    normed : array, shape (N_genes, N_samples)
+        The RPKM normalized counts matrix.
+    """
     N = np.sum(counts, axis=0)  # sum each column to get total reads per sample
     L = lengths
     C = counts
 
-    rpkm = ( (10e9 * C) / N[np.newaxis, :] ) / L[:, np.newaxis]
+    normed = 1e9 * C / (N[np.newaxis, :] * L[:, np.newaxis])
```

Comment:
Double checking that this should have changed: 10e9 -> 1e9 and `/ L` became `* L`